### PR TITLE
[DOC] update doc threshold image

### LINKED ISSUE #2048: Documentation of threshold_img unclear
1.) From the documentation of `threshold_img` it's unclear whether features below threshold will be kept or just set to zero. When I first saw the documentation I thought that `threshold_img` is a feature reduction procedure and will return a feature reduced subset. I had to find out by myself that features below threshold will be set to zero. The used phrase _"will be kept"_ also leads to the false idea that `threshold_img` is a feature reduction procedure.

2.) If I read the source code right `img_data[np.abs(img_data) < cutoff_threshold] = 0.)`, all features below threshold will be set to zero. This logically implies that all other values greater **or equal** to threshold will be kept. Therefore, the sentence 

> _The voxels which have intensities greater than this score will be kept._ 

should be changed to

>  _The voxels which have intensities greater or equal than this score will be kept._

3.) It's very nice that `threshold_img` can also handle percentiles because otherwise you always have to inspect the data first (or look at descriptive stats) to find a good voxel intensity value as threshold. However it would be nice to give an explanation of how the percentile rank is calculated behind the scenes. Again if I read the source code right the given percentile string (e.g. '80%') is converted to a percentile rank using `scipy.stats.scoreatpercentile`. It would be nice to give a reference to this function and to document which keyword-arguments of `scipy.stats.scoreatpercentile` are used when `check_threshold` is called.

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #2048

Changes proposed in this pull request:

1. Clarify that voxels with values
  - inferior to threshold are set to 0
  - supeerior or equal to threshold are kept as is

2. Explain how percentiles are computed.

### CODE REVIEW

File: nilearn/image/image.py [hunk 1]
```diff
@@ -861,14 +861,18 @@ def threshold_img(
...
-        If string, it should finish with percent sign e.g. "80%" and we threshold
-        based on the score obtained using this percentile on the image data. The
-        voxels which have intensities greater than this score will be kept.
+        Voxels with intensities less than the requested threshold
+        will be set to zero.
+        Those with intensities greater or equal than the requested threshold
+        will keep their original value.
+        If float, we threshold the image based on image intensities.
+        The given value should be within the range of minimum and maximum intensity
+        of the input image.
```
Comment (ymzayek) [inline nilearn/image/image.py:870]:
```suggestion
        The given value should be within the range of minimum and maximum
        intensity of the input image.
```

File: nilearn/image/image.py [hunk 2]
```diff
@@ -861,14 +861,18 @@ def threshold_img(
...
+        If string, it should finish with percent sign e.g. "80%"
+        and we threshold based on the score obtained
+        using this percentile on the image data.
         The given string should be within the range of "0%" to "100%".
+        The percentile rank is computed using ``scipy.stats.scoreatpercentile``.
```
Comment (ymzayek) [inline nilearn/image/image.py:875]:
```suggestion
        The percentile rank is computed using
        :func:`scipy.stats.scoreatpercentile`.
```

**Review by ymzayek — COMMENTED** [review]
LGTM @Remi-Gau! Just minor things. I'd also add a whatsnew entry.

**Comment by Remi-Gau** [comment]
> I'd also add a whatsnew entry.

added something in latest.rst.

File: doc/changes/latest.rst [hunk 3]
```diff
@@ -21,3 +21,8 @@ Enhancements
 
 Changes
 -------
+
+- The documentation for :func:`~image.threshold_img` has been improved,
```
Comment (ymzayek) [inline doc/changes/latest.rst:25]: In general we have the convention of one line per entry in this file (see [PR Structure](https://nilearn.github.io/stable/development.html#pr-structure)). I know this is broken sometimes and I don't have a strong opinion about it but better to try to keep things consistent.

**Review by ymzayek — APPROVED** [review]
LGTM @Remi-Gau

File: doc/changes/latest.rst (hunk 3 above)
Comment (Remi-Gau) [reply doc/changes/latest.rst:25]: woopsy

I am trying to use [semantic linebreaks](https://sembr.org/) more and more, so I completely overlooked this.

To make it harder to miss, would it make sense to add links to the relevant headings in the [PR checklist](https://github.com/nilearn/nilearn/blob/b7e5efdd37a6b4cc276763fc5f2a2cde81f7af73/.github/workflows/auto-comment.yml#L22) ?

Comment (ymzayek) [reply doc/changes/latest.rst]: Yes I think that's a good idea! Can you open an issue for that?

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by ymzayek** [comment]
Merging

**Pull request merged**